src/modules/siptrace/siptrace.c

```diff
@@ -304,10 +338,30 @@ static int mod_init(void)
 	/* register callbacks to TM */
 	if(load_tm_api(&tmb) != 0) {
 		LM_WARN("can't load tm api. Will not install tm callbacks.\n");
-	} else if(tmb.register_tmcb(0, 0, TMCB_REQUEST_IN, trace_onreq_in, 0, 0)
-			  <= 0) {
-		LM_ERR("can't register trace_onreq_in\n");
-		return -1;
+	}
+
+	if (load_dlg_api(&dlgb) < 0) {
+		LM_WARN("can't load dlg api. Will not install dialog callbacks.\n");
```

Comment:
Below you log on ERROR level if the dialog tracing is not available, makes probably sense to log here with ERROR as well.

Reply:
I'm not totally sure I understand correctly. My intention was the module not to depend on either tm or dialog, that's why I put LM_WARN there. That's because it can and should be used in stateless mode without any issues. Is that what you're referring to?

Reply:
It was only a minor comment. But having though about one more time its makes sense now with your comment.